src/components/Carousel.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from "react";
+import Slider from "react-slick";
+import './style/Carousel.scss';
+import f1 from '../img/frames/0.png';
+import f2 from '../img/frames/1.png';
+import f3 from '../img/frames/2.png';
+import f4 from '../img/frames/3.png';
+import f5 from '../img/frames/4.png';
+import f6 from '../img/frames/5.png';
+import f7 from '../img/frames/6.png';
+import f8 from '../img/frames/7.png';
+import f9 from '../img/frames/8.png';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlay } from '@fortawesome/free-solid-svg-icons';
+import { faPause } from '@fortawesome/free-solid-svg-icons';
+
+export default class AutoPlayMethods extends Component {
+  constructor(props) {
+    super(props);
+    this.play = this.play.bind(this);
+    this.pause = this.pause.bind(this);
+  }
+  play() {
+    this.slider.slickPlay();
+  }
+  pause() {
+    this.slider.slickPause();
+  }
+  render() {
+    const settings = {
+      dots: true,
+      fade: true,
+      infinite: true,
+      slidesToShow: 1,
+      slidesToScroll: 1,
+      autoplay: true,
+      autoplaySpeed: 2000,
+      arrows: false
+    };
+    return (
+      <div className = "main">
+        <Slider ref={slider => (this.slider = slider)} {...settings}>
```

Comment:
This is the part that irked me: `this.slider = slider`

Reply:
What is generally the problem with inline styling? Does it take too long to render for older browsers/poorer connections? 

And if I understand correctly, the alternate route is making a carousel, setting all the pictures' opacity to 0, and then making each picture's opacity 1 after a particular interval?

Also, one final question about the syntax: I don't understand the props.children.map => <div> syntax here 
```
        {props.children.map((c, i) => <div key={i} className={`carousel-slide ${i === currentSlide ? 'visible' : ''}`}>{c}</div>)}
```

Reply:
> What is generally the problem with inline styling? Does it take too long to render for older browsers/poorer connections?

There actually isn't much a problem with inline styling! I only pointed it out because it is something that is indicative of some dicey code. Inline styles override any existing ones in your stylesheets, and often lead to unpredictable problems between components.

> And if I understand correctly, the alternate route is making a carousel, setting all the pictures' opacity to 0, and then making each picture's opacity 1 after a particular interval?

Yes, the way that the example carousel works is by setting the opacity of every picture to zero by default (gives every child element the `.carousel-slide` class), then applies the `.visible` class to the "current slide", which is determined by the state variable `currentSlide`. We then leverage `useEffect` to set an interval that increments that `currentSlide` state variable by one every handful of milliseconds, wrapping around to zero when we exceed the number of children passed to the component.

The kicker here is that we draw all the elements on top of each other by adding a container class that positions all child elements (`.carousel > *`) absolutely in the same location.

> I don't understand the props.children.map =>
> syntax here

This is a pretty dense line, so I'll break it up into a more verbose set and comment each one. Let me know if there's anything that's unclear!

```js
import React, { useEffect, useState } from 'react';
import './Example.css';

export default function Carousel(props) {
  const [ currentSlide, setCurrentSlide ] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentSlide((currentSlide + 1) % props.children.length);
    }, 1000);

    return () => clearInterval(interval);
  });

  // Every React component has a property passed down to it silently:
  // props.children.
  // This is an array of all the **child** components passed to it.
  // e.g.:
  //
  // <Carousel>
  //   <h1>First child</h1>
  //   <h2>Second child</h2>
  // </Carousel>
  //
  // Here, our children are the <h1/> and <h2/> tags.

  // Let's create an Array to hold our mutated children.
  let newChildren = [];

  // Now, we loop through props.children and mutate each
  // component.
  for (let i = 0; i < props.children.length; i++) {
    // Our current child element will be 'c'.
    const c = props.children[i];

    // We need to create a class string. Each component
    // needs to have class 'carousel-slide', though, so
    // we start from there.
    let classString = 'carousel-slide';

    // If our current child that we're iterating through
    // is the current slide, it should have the 'visible'
    // class. Append it with a leading space (to separate
    // class names).
    if (i === currentSlide)
      classString += ' visible';

    // Then, construct our child and push it onto our list.
    newChildren.push(
      // Each child needs to be wrapped with another div with
      // our provided classes.
      // Since it is a member of an array, it also needs a unique
      // 'key' prop. We can just use the index of the item.
      <div key={i} className={classString}>
        {c}
      </div>
    )
  }
  
  return (
    <div className='carousel'>
      {newChildren}
    </div>
  );
}
```

Reply:
I understand now, thank you so much!

Reply:
Is it alright to stick to react-slick for now and change it later if there are any issues? Because I'm currently working on making new frames (the current ones are slightly inconsistent)

Reply:
If you'd like to stick with `react-slick`, that's alright, but I would discourage it.

The main point of my example here was to point out that making your own isn't as gnarly as it sounds, and offers more control over the process without pulling some alarming cards.

Reply:
I can definitely work on the carousel alongside but don't know if I'll have the time for it since I'm working on new frames for the carousel. I'll try my best to switch away from react-slick